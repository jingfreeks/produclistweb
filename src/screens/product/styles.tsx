import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  justify-content: center;
  
`;

export const LoadingContainerStyled = styled(Container)`
  display: flex;
  
  justify-content: center;
  height: 100vh;
`;
