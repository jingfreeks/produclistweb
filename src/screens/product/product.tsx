import React, { useCallback, Suspense } from "react";
import { Container,LoadingContainerStyled } from "./styles";
import { useGetProductQuery } from "../../slice/producSlice";
import { ListItem, } from "./component";
const Product = () => {
  const {
    data: product,
    isLoading,
  } = useGetProductQuery<{
    refetch: () => void;
    data: any;
    isLoading: boolean;
    isSuccess: string;
    isError: boolean;
    error: any;
  }>("getProduct");

  const renderProduct = useCallback(() => {
    let productItem = product;
    return product?.ids.map((item: string) => {
      return <ListItem key={item} item={productItem?.entities[item]} />;
    });
  }, [product]);
  const Loading = () => {
    return (
      <LoadingContainerStyled>
        <h2>ðŸŒ€ Loading...</h2>
      </LoadingContainerStyled>
    );
  };
  if (isLoading) {
    return <Loading />;
  }
  return (
    <Container>
      <div>
        <ul>{renderProduct()}</ul>
      </div>
    </Container>
  );
};

export default Product;
