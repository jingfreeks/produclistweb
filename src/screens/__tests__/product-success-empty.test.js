import { renderWithProviders } from "../../slice/testframework";
import {Product} from "../product";
jest.mock('../../slice/producSlice',()=>({
    useGetProductQuery:()=>({
      isLoading:false,
      isSuccess:true,
      isError:false,
      data:{ids:[]}
    })
  }))
// Mock useRouter:
describe("Product Success Empty data ", () => {
  it("renders Layout unchanged to match snapshot", () => {
    const { container } = renderWithProviders(<Product />);
    expect(container).toMatchSnapshot();
  });
});
