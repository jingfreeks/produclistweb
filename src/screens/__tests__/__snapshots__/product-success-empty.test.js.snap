// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Success Empty data  renders Layout unchanged to match snapshot 1`] = `
<div>
  <div
    class="sc-blHHSb jzxxPQ"
  >
    <div>
      <ul />
    </div>
  </div>
</div>
`;
